src/Compilers/CSharp/Portable/Symbols/Source/SourceTypeParameterSymbol.cs

```diff
@@ -267,6 +267,29 @@ private void CheckConstraintTypeConstraints(DiagnosticBag diagnostics)
             }
         }
 
+        private void CheckUnmanagedConstraint(DiagnosticBag diagnostics)
+        {
+            if (this.HasUnmanagedTypeConstraint)
+            {
+                bool modifyCompilation;
+
+                switch (this.ContainingSymbol)
+                {
+                    case SourceOrdinaryMethodSymbol _:
+                    case SourceMemberContainerTypeSymbol _:
+                        modifyCompilation = true;
+                        break;
+                    case LocalFunctionSymbol _:
+                        modifyCompilation = false;
+                        break;
+                    default:
+                        throw ExceptionUtilities.UnexpectedValue(this.ContainingSymbol);
```

Comment:
>throw ExceptionUtilities.UnexpectedValue(this.ContainingSymbol); [](start = 24, length = 64)

Not sure if throwing is absolutely necessary here. For example we don't do this when we call LoadAndValidateAttributes where container a Local Function makes a difference. I'd say a simple check if container is a Local Function would be good enough.

Reply:
The question is should we throw in case there is a unmanaged symbol that we didn't account for/test. I'd say let's throw here for debugging reasons. If this was true for a new symbol accidently, it might just end up crashing during emit, with a harder trace to debug.

---
In reply to: [177923556](https://github.com/dotnet/roslyn/pull/25680#discussion_r177923556) [](ancestors = 177923556)